### PR TITLE
fatfs: Define new system type for FDPP

### PR DESCRIPTION
**Pull request by andrewbird**
Use a destinct filename for FDPP boot file. This allows us to do the
right thing if one is present and FDPP is not compiled in.

BTW is filename 'FDPPBOOT.SYS' okay?

### CODE REVIEW

File: src/base/misc/fatfs.c [hunk 1]
```diff
@@ -666,6 +668,8 @@ static char *system_type(unsigned int t) {
         return "Old FreeDOS";
     case FD_D:
         return "FreeDOS";
+    case FDP_D:
+        return "FreeDOS Plugin";
```
Comment (stsp) [inline src/base/misc/fatfs.c]: No, please don't call it freedos. ;)
Its a separate project, and "PP" stands
for "PlusPlus" (because of C++ port).

**Comment by stsp** [comment]
> BTW is filename 'FDPPBOOT.SYS' okay?

fdppkrnl.sys sounds better.

**Comment by andrewbird** [comment]
I'm confused by what's in the kernel.sys, but no matter.

**Comment by stsp** [comment]
> I'm confused by what's in the kernel.sys, but no matter.

In fact, everything that you can find in the `*.asm`
files. So a rather large chunk of the kernel.
I did a small change to "sfs" variable setup, could
you please adjust your patch?
Also it would be nice to have the fdpp counterpart
patch so that things still work.

**Comment by andrewbird** [comment]
here you go

**Comment by stsp** [comment]
Yes but how about the fdpp commit of the same thing?

**Comment by andrewbird** [comment]
Any way to tell in the FDPP makefile that we are compiling for libfdpp.so / and the fdppkrnl.sys file, otherwise it'll just be installing with the new name?

**Comment by stsp** [comment]
What do you mean?
It doesn't install the kernel.sys because it
doesn't know where.

**Comment by andrewbird** [comment]
The make install target copies it to the bin directory

**Comment by stsp** [comment]
Well, install target doesn't do this, but `production`
target does. I added the patch to change the name
in `bin`.

**Comment by stsp** [comment]
Thanks!

**Comment by andrewbird** [comment]
ahh you beat me to it.
~~~
                                                                                
../bin/$(TARGET).sys: kernel.sys                                                
                $(CP) kernel.sys ..$(DIRSEP)bin                                 
                $(CP) kernel.sys ..$(DIRSEP)bin$(DIRSEP)$(TARGET).sys           
                $(CP) kernel.map ..$(DIRSEP)bin$(DIRSEP)$(TARGET).map           
 ~~~

**Comment by stsp** [comment]
I think this part is done to allow multiple
builds by different compilers. The TARGET
was coding a compiler name in 3 letters,
CPU model, fat32 etc. But why not to abuse
it - this won't break the multiple builds anyway,
as no one will clash with fdppkrnl.sys. :)

**Pull request merged**